cmd/nerdctl/run_security_linux_test.go

```diff
@@ -190,3 +191,37 @@ func TestRunSeccompCapSysPtrace(t *testing.T) {
 	base.Cmd("run", "--rm", "--cap-add", "sys_ptrace", testutil.AlpineImage, "sh", "-euxc", "apk add -q strace && strace true").AssertOK()
 	// Docker/Moby 's seccomp profile allows ptrace(2) by default, but containerd does not (yet): https://github.com/containerd/containerd/issues/6802
 }
+
+func TestRunPrivileged(t *testing.T) {
+	// docker does not support --privileged-without-host-devices
+	testutil.DockerIncompatible(t)
+
+	base := testutil.NewBase(t)
+
+	devPath := "/dev/dummy-zero"
+
+	// a dummy zero device: mknod /dev/dummy-zero c 1 5
+	helperCmd := exec.Command("mknod",
+		[]string{devPath, "c", "1", "5"}...)
+	if out, err := helperCmd.CombinedOutput(); err != nil {
+		err = fmt.Errorf("cannot create %q: %q: %w", devPath, string(out), err)
+		t.Fatal(err)
+	}
+
+	// ensure the file will be removed in case of failed in the test
+	defer func() {
+		exec.Command("rm", devPath).Run()
+	}()
+
+	// get device with host devices
+	base.Cmd("run", "--rm", "--privileged", testutil.AlpineImage, "ls", devPath).AssertOutExactly(devPath + "\n")
+
+	// get device without host devices
+	res := base.Cmd("run", "--rm", "--privileged-without-host-devices", testutil.AlpineImage, "ls", devPath).Run()
```

Comment:
Should `ls /dev` return empty list in this case, would that be a better test?

Reply:
No, this option is only disable passing all host devices to containers. Containers will have its default devices [automatically created](https://github.com/opencontainers/runc/blob/d11f4d756e85ece5cdba8bb69f8bd4db3cdcbeab/libcontainer/specconv/spec_linux.go#L173-L174).